## If, elif, else – for – while – operadores – list comprehensions
LingProg - Lukas Barbosa de Oliveira - RA: 81714448 <br><br>

1. Faça um Programa que peça dois números e imprima o maior deles.

In [1]:
n1 = int(input('Insira um número: '))
n2 = int(input('Insira um outro número: '))

if n1>n2:
    print(f'O maior número é o {n1}')
elif n2>n1:
    print(f'O maior número é o {n2}')
else:
    print('Números iguais')
    

Insira um número: 90
Insira um outro número: 67
O maior número é o 90


<b>2. Faça um Programa que verifique se uma letra digitada é vogal ou consoante.</b>

In [5]:
vogais = 'aeiou'
letra = input('digite uma letra: ')

if letra in vogais:
    print(f'A letra {letra} é uma vogal')
else:
    print(f'A letra {letra} é uma consoante')

digite uma letra: u
A letra u é uma vogal


<b>3. Faça um programa para a leitura de duas notas parciais de um aluno. O programa
deve calcular a média alcançada por aluno e apresentar:</b><br>
- A mensagem "Aprovado", se a média alcançada for maior ou igual a sete;<br>
- A mensagem "Reprovado", se a média for menor do que sete;<br>
- A mensagem "Aprovado com Distinção", se a média for igual a dez.

In [12]:
nota1 = float(input('Insira a primeira nota parcial: '))
nota2 = float(input('Insira a segunda nota parcial: '))

media = (nota1+nota2)/2

if media >= 7 and media < 10:
    print(f'Média: {media} - APROVADO')
elif media < 7:
    print(f'Média: {media} - REPROVADO')
elif media == 10:
    print(f'Média: {media} - Aprovado com Distinção')


Insira a primeira nota parcial: 5
Insira a segunda nota parcial: 10
Média: 7.5 - APROVADO


<b>4. Faça um Programa que leia três números e mostre-os em ordem decrescente.</b>

In [20]:
n41 = int(input('Insira um número: '))
n42 = int(input('Insira um segundo número: '))
n43 = int(input('Insira um terceiro número: '))

if n41 > n42 and n41 >n43:
    if n42 > n43:
        print(f'Ordem decrecente: {n41} {n42} {n43}')
    else:
        print(f'Ordem decrecente: {n41} {n43} {n42}')
        
elif n42 > n41 and n42 >n43:
    if n41 > n43:
        print(f'Ordem decrecente: {n42} {n41} {n43}')
    else:
        print(f'Ordem decrecente: {n42} {n43} {n41}')
        
elif n43 > n41 and n43 >n42:
    if n41 > n42:
        print(f'Ordem decrecente: {n43} {n41} {n42}')
    else:
        print(f'Ordem decrecente: {n43} {n42} {n41}')

Insira um número: 58
Insira um segundo número: 34
Insira um terceiro número: 68
Ordem decrecente: 68 58 34


<b>
5: As Organizações Tabajara resolveram dar um aumento de salário aos seus colaboradores e lhe contrataram para desenvolver o programa que calculará os reajustes. Faça um programa que recebe o salário de um colaborador e o reajuste segundo o seguinte critério, baseado no salário atual:</b><br>

 salários até RS 280,00 (incluindo) : aumento de 20%<br>
 salários entre RS 280,00 e RS 700,00 : aumento de 15%<br>
 salários entre RS 700,00 e RS 1500,00 : aumento de 10%<br>
 salários de RS 1500,00 em diante : aumento de 5porcento<br>
Após o aumento ser realizado, informe na tela:<br><br>

- o salário antes do reajuste;<br>
- o percentual de aumento aplicado;<br>
- o valor do aumento;<br>
- o novo salário, após o aumento.

In [21]:
salario = float(input('Insira o salário: '))
if salario <= 280:
    percentual = 0.20
elif salario > 280 and salario <= 700:
    percentual = 0.15
elif salario > 700 and salario <= 1500:
    percentual = 0.10
else:
    percentual = 0.05
novoSalario = salario + (salario * percentual)

print('Salário antes do reajuste:', salario)
print(f'Percentual de aumento aplicado: {percentual * 100}%')
print('Valor do aumento:', salario * percentual)
print('Novo salário:', novoSalario)

Insira o salário: 800
Salário antes do reajuste: 800.0
Percentual de aumento aplicado: 10.0%
Valor do aumento: 80.0
Novo salário: 880.0


<b>6. Faça um Programa que leia um número e exiba o dia correspondente da semana.
(1-Domingo, 2- Segunda, etc.), se digitar outro valor deve aparecer valor inválido.
</b>

In [11]:
dia6 = int(input('Digite um número de 1 a 7: '))
if dia6 == 1:
    print('Domingo')
elif dia6 == 2:
     print('Segunda')
elif dia6 == 3:
     print('Terça')
elif dia6 == 4:
     print('Quarta')
elif dia6 == 5:
     print('Quinta')
elif dia6 == 6:
     print('Sexta')
else:
     print('Valor inválido')

Digite um número de 1 a 7: 2
Segunda


<b>7. Faça um programa que lê as duas notas parciais obtidas por um aluno numa
disciplina ao longo de um semestre, e calcule a sua média. A atribuição de conceitos
obedece à tabela abaixo:</b><br><br>
 Média de Aproveitamento Conceito<br>
 Entre 9.0 e 10.0 A<br>
 Entre 7.5 e 9.0 B<br>
 Entre 6.0 e 7.5 C<br>
 Entre 4.0 e 6.0 D<br>
 Entre 4.0 e zero E<br><br>
O algoritmo deve mostrar na tela as notas, a média, o conceito correspondente e a
mensagem “APROVADO” se o conceito for A, B ou C ou “REPROVADO” se o conceito
for D ou E.

In [13]:
nota16 = float(input('Insira a primeira nota: '))
nota26 = float(input('Insira a segunda nota: '))
media6 = (nota16 + nota26) / 2

if media6 >= 9 and media6 <= 10:
    conceito = 'A'
elif media6 >= 7.5 and media6 < 9:
    conceito = 'B'
elif media6 >= 6 and media6 < 7.5:
    conceito = 'C'
elif media6 >= 4 and media6 < 6:
    conceito = 'D'
elif media6 < 4:
    conceito = 'E'
    
print('Nota 1: ', nota16)
print('Nota 2: ', nota26)
print('Média: ', media6)
print('Conceito: ', conceito)

if conceito == "A":
    print("Aluno - bAPROVADO")
elif conceito == "B":
    print("Aluno- APROVADO")
elif conceito == "C":
    print("Aluno - APROVADO")
else:
    print("Aluno - REPROVADO")

Insira a primeira nota: 5
Insira a segunda nota: 9
Nota 1:  5.0
Nota 2:  9.0
Média:  7.0
Conceito:  C
Aluno - APROVADO


<b>8: Faça um Programa que peça os 3 lados de um triângulo. O programa deverá informar se os valores podem ser um triângulo. Indique, caso os lados formem um triângulo, se o mesmo é: equilátero, isósceles ou escaleno. Dicas:</b>
<br><br>
Três lados formam um triângulo quando a soma de quaisquer dois lados for maior<br>
que o terceiro;
 Triângulo Equilátero: três lados iguais;<br>
Triângulo Isósceles: quaisquer dois lados iguais;<br>
 Triângulo Escaleno: três lados diferentes;<br>

In [19]:
l1 = float(input('Insira o primeiro lado: '))
l2 = float(input('Insira o segundo lado: '))
l3 = float(input('Insira o terceiro lado: '))

if l1 < l2 + l3 and l2 < l1 + l3 and l3 < l1 + l2:
    print('Os valores dos lados informados formam um triângulo:')
    if l1 == l2 == l3:
        print('Equilátero.')
    elif l1 == l2 or l1 == l3 or l2 == l3:
        print('Isósceles.')
    else:
        print('Escaleno.')
else:
    print('Os valores dos lados informados não formam um triângulo.')

Insira o primeiro lado: 3
Insira o segundo lado: 5
Insira o terceiro lado: 6
Os valores dos lados informados formam um triângulo:
Escaleno.


<b>9. Faça um programa que calcule as raízes de uma equação do segundo grau, na
forma ax2 + bx + c. O programa deverá pedir os valores de a, b e c e fazer as
consistências, informando ao usuário nas seguintes situações:</b><br>
<br>- Se o usuário informar o valor de A igual a zero, a equação não é do segundo grau e
o programa não deve fazer pedir os demais valores, sendo encerrado;
<br>- Se o delta calculado for negativo, a equação não possui raizes reais. Informe ao
usuário e encerre o programa;
<br>- Se o delta calculado for igual a zero a equação possui apenas uma raiz real;
informe-a ao usuário;
<br>- Se o delta for positivo, a equação possui duas raiz reais; informe-as ao usuário;

In [21]:
import math

a = float(input('Insira o A: '))
if a == 0:
    print("Não é uma equação de segundo grau, programa encerrado.")
else:
    b = float(input('Insira o B: '))
    c = float(input('Insira o C: '))
    delta = b**2 - 4*a*c
    if delta < 0:
        print("A equação não possui raizes reais, programa encerrado")
    elif delta == 0:
        x = -b / (2*a)
        print("A equação possui apenas uma raiz real:", x)
    else:
        x1 = (-b + math.sqrt(delta)) / (2*a)
        x2 = (-b - math.sqrt(delta)) / (2*a)
        print(f"A equação possui duas raiz reais: {x1} e {x2}")

Insira o A: -1
Insira o B: 2
Insira o C: 3
A equação possui duas raiz reais: -1.0 e 3.0


<b>10. Faça um Programa para um caixa eletrônico. O programa deverá perguntar ao
usuário a valor do saque e depois informar quantas notas de cada valor serão
fornecidas. As notas disponíveis serão as de 1, 5, 10, 50 e 100 reais. O valor mínimo é
de 10 reais e o máximo de 600 reais. O programa não deve se preocupar com a
quantidade de notas existentes na máquina.</b><br>
<br>- Exemplo 1: Para sacar a quantia de 256 reais, o programa fornece duas notas de
100, uma nota de 50, uma nota de 5 e uma nota de 1;
<br>- Exemplo 2: Para sacar a quantia de 399 reais, o programa fornece três notas de
100, uma nota de 50, quatro notas de 10, uma nota de 5 e quatro notas de 1.

In [22]:
saque = int(input("Informe o valor a ser sacado: "))
if saque >= 10 and saque <= 600:
    n100 = saque // 100
    resto = saque % 100
    print(f'{n100} notas de 100 reais;')
    n50 = resto // 50
    resto = resto % 50
    print(f'{n50} notas de 50 reais;')
    n10 = resto // 10
    resto = resto % 10
    print(f'{n10} notas de 10 reais;')
    n5 = resto // 5
    resto = resto % 5
    print(f'{n5} notas de 5 reais;')
    print(f'{resto} notas de 1 real.')
else:
    print("Valor não permitido.")

Informe o valor a ser sacado: 452
4 notas de 100 reais;
1 notas de 50 reais;
0 notas de 10 reais;
0 notas de 5 reais;
2 notas de 1 real.


<b>11. Faça um programa que faça 5 perguntas para uma pessoa sobre um crime. As
perguntas são:</b><br>
<br>"Telefonou para a vítima?"
<br>"Esteve no local do crime?"
<br>"Mora perto da vítima?"
<br>"Devia para a vítima?"
<br>"Já trabalhou com a vítima?"<br>
<br>O programa deve no final emitir uma classificação sobre a participação da pessoa no
crime. Se a pessoa responder positivamente a 2 questões ela deve ser classificada
como "Suspeita", entre 3 e 4 como "Cúmplice" e 5 como "Assassino". Caso contrário,
ele será classificado como "Inocente".

In [23]:
score = 0
respostas = [
    input("Telefonou para a vítima? (s/n): "),
    input("Esteve no local do crime? (s/n): "),
    input("Mora perto da vítima? (s/n): "),
    input("Devia para a vítima? (s/n): "),
    input("Já trabalhou com a vítima? (s/n): ")
]

for resposta in respostas:
    if resposta == "s":
        score += 1
        
if score == 2:
    print("Suspeita")
elif score >= 3 and score <= 4:
    print("Cúmplice")
elif score == 5:
    print("Assassino")
else:
    print("Inocente")

Telefonou para a vítima? (s/n): s
Esteve no local do crime? (s/n): n
Mora perto da vítima? (s/n): s
Devia para a vítima? (s/n): s
Já trabalhou com a vítima? (s/n): n
Cúmplice


<b>12. Uma fruteira está vendendo frutas com a seguinte tabela de preços:</b><br>
<br> Até 5 Kg Acima de 5 Kg
<br>Morango RS 2,50 por Kg RS 2,20 por Kg
<br>Maçã RS 1,80 por Kg RS 1,50 por Kg
<br>Se o cliente comprar mais de 8 Kg em frutas ou o valor total da compra ultrapassar
R$ 25,00, receberá ainda um desconto de 10porcento sobre este total. Escreva um
algoritmo para ler a quantidade (em Kg) de morangos e a quantidade (em Kg) de
maças adquiridas e escreva o valor a ser pago pelo cliente.


In [ ]:
preco = 0
kgMorg = float(input("Insira a quantidade em kg de morangos: "))
kgMaca = float(input("Insira a quantidade em kg de maçãs: "))

if kgMorg < 5:
    preco += kgMorg * 2.50
else:
    preco += kgMorg * 2.20
    
if kgMaca < 5:
    preco += kgMaca * 1.80
else:
    preco += kgMaca * 1.50
    
if preco > 25.00:
    preco *= 0.90

print('Valor a ser pago: ', preco)

<b>13. Faça um programa que peça uma nota, entre zero e dez. Mostre uma mensagem
caso o valor seja inválido e continue pedindo até que o usuário informe um valor
válido.</b>

In [25]:
num = int(input("Insira uma nora entre zero e dez: "))
while num < 0 or num > 10:
    print("Valor inválido.")
    num = int(input("Insira uma nora entre zero e dez: "))
else:
    print("Valor válido.")

Insira uma nora entre zero e dez: 12
Valor inválido.
Insira uma nora entre zero e dez: 12
Valor inválido.
Insira uma nora entre zero e dez: 12
Valor inválido.
Insira uma nora entre zero e dez: 4
Valor válido.


<b>13 (V2??) . Faça um programa que leia e valide as seguintes informações: </b><br>
Nome: maior que 3 caracteres;<br>
Idade: entre 0 e 150;<br>
Salário: maior que zero;<br>
Sexo: 'f' ou 'm';<br>
Estado Civil: 's', 'c', 'v', 'd';<br>

In [26]:
nome = input("Insira o nome (maior que três caracteres): ")
while len(nome) <= 3:
    nome = input("Insira o nome (maior que três caracteres): ")
    
idade = int(input("Insira a idade (0 - 150): "))
while idade < 0 or idade > 150:
    idade = int(input("Insira a idade (0 - 150): "))
    
salario = int(input("Insira o salário (maior que zero): "))
while salario <= 0:
    salario = int(input("Insira o salário (maior que zero): "))
    
sexo = input("Insira o sexo (f/m): ")
while not (sexo in ['f', 'm']):
    sexo = input("Insira o sexo (f/m): ")
    
estadocivil = input("Insira o estado civil (s/c/v/d): ")
while not (estadocivil in ['s', 'c', 'v', 'd']):
    estadocivil = input("Insira o estado civil (s/c/v/d): ")

Insira o nome (maior que três caracteres): lukas
Insira a idade (0 - 150): 18
Insira o salário (maior que zero): 200
Insira o sexo (f/m): m
Insira o estado civil (s/c/v/d): s


<b>14. Faça um programa que leia 5 números e informe o maior número.</b>

In [27]:
maior = int(input("Digite o 1º número: "))
for num in range(2,6):
    x = int(input(f"Digite o {num}º número: "))
    if x > maior:
        maior = x
print("Maior:", maior)

Digite o 1º número: 4
Digite o 2º número: 5
Digite o 3º número: 67
Digite o 4º número: 42
Digite o 5º número: 4
Maior: 67


<b>15. Faça um programa que leia 5 números e informe a soma e a média dos números.</b>

In [29]:
soma = 0
for num in range(1,6):
    soma += int(input(f"Digite o {num}º número: "))
print("Soma:", soma)
print("Media:", soma / 5)

Digite o 1º número: 65
Digite o 2º número: 47
Digite o 3º número: 47
Digite o 4º número: 69
Digite o 5º número: 321
Soma: 549
Media: 109.8


<b>16. Faça um programa que imprima na tela apenas os números ímpares entre 1 e 50</b>

In [30]:
for num in range(1, 51):
    if num % 2 != 0:
        print(num)

1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49


<b>17. Desenvolva um gerador de tabuada, capaz de gerar a tabuada de qualquer
número inteiro entre 1 a 10. O usuário deve informar de qual numero ele deseja ver
a tabuada. A saída deve ser conforme o exemplo abaixo:</b><br>
<br>Tabuada de 5:
<br>5 X 1 = 5
<br>5 X 2 = 10
<br>...
<br>5 X 10 = 50

In [31]:
tabuada = int(input("Insira o número da tabuada: "))
for mt in range(1, 11):
    res = mt*tabuada
    print(f"{tabuada} X {mt} = {res}")

Insira o número da tabuada: 7
7 X 1 = 7
7 X 2 = 14
7 X 3 = 21
7 X 4 = 28
7 X 5 = 35
7 X 6 = 42
7 X 7 = 49
7 X 8 = 56
7 X 9 = 63
7 X 10 = 70


<b>18. A série de Fibonacci é formada pela seqüência 1,1,2,3,5,8,13,21,34,55,... Faça um
programa capaz de gerar a série até o n−ésimo termo.</b>

In [34]:
a = 0
b = 1
nesimo = int(input("Até qual termo? "))
for i in range(nesimo):
    if a != 0:
        print(a)
    aux = b
    b = a + b
    a = aux

Até qual termo? 10
1
1
2
3
5
8
13
21
34


<b>19. Faça um programa que calcule o fatorial de um número inteiro fornecido pelo
usuário. Ex.: 5!=5.4.3.2.1=120</b>

In [36]:
fatorial = int(input("Fatorial de: "))
for i in range(1, fatorial):
    fatorial = fatorial * i
print(fatorial)

Fatorial de: 4
24


<b>20. O Sr. Manoel Joaquim possui uma grande loja de artigos de RS 1,99, com cerca
de 10 caixas. Para agilizar o cálculo de quanto cada cliente deve pagar ele
desenvolveu um tabela que contém o número de itens que o cliente comprou e ao
lado o valor da conta. Desta forma a atendente do caixa precisa apenas contar
quantos itens o cliente está levando e olhar na tabela de preços. Você foi contratado
para desenvolver o programa que monta esta tabela de preços, que conterá os
preços de 1 até 50 produtos, conforme o exemplo abaixo:</b><br>
<br>Lojas Quase Dois - Tabela de preços
<br>1 - RS 1.99
<br>2 - RS 3.98
<br>...
<br>50 - RS 99.50


In [37]:
for num in range(1, 51):
    preco = num * 1.99
    print(f"{num} - R$ {preco}")

1 - R$ 1.99
2 - R$ 3.98
3 - R$ 5.97
4 - R$ 7.96
5 - R$ 9.95
6 - R$ 11.94
7 - R$ 13.93
8 - R$ 15.92
9 - R$ 17.91
10 - R$ 19.9
11 - R$ 21.89
12 - R$ 23.88
13 - R$ 25.87
14 - R$ 27.86
15 - R$ 29.85
16 - R$ 31.84
17 - R$ 33.83
18 - R$ 35.82
19 - R$ 37.81
20 - R$ 39.8
21 - R$ 41.79
22 - R$ 43.78
23 - R$ 45.77
24 - R$ 47.76
25 - R$ 49.75
26 - R$ 51.74
27 - R$ 53.73
28 - R$ 55.72
29 - R$ 57.71
30 - R$ 59.7
31 - R$ 61.69
32 - R$ 63.68
33 - R$ 65.67
34 - R$ 67.66
35 - R$ 69.65
36 - R$ 71.64
37 - R$ 73.63
38 - R$ 75.62
39 - R$ 77.61
40 - R$ 79.6
41 - R$ 81.59
42 - R$ 83.58
43 - R$ 85.57
44 - R$ 87.56
45 - R$ 89.55
46 - R$ 91.54
47 - R$ 93.53
48 - R$ 95.52
49 - R$ 97.51
50 - R$ 99.5


<b>21. O Sr. Manoel Joaquim acaba de adquirir uma panificadora e pretende implantar
a metodologia da tabelinha, que já é um sucesso na sua loja de 1,99. Você foi
contratado para desenvolver o programa que monta a tabela de preços de pães, de
1 até 50 pães, a partir do preço do pão informado pelo usuário, conforme o exemplo
abaixo:</b><br>
<br>Preço do pão: RS 0.18
<br>Panificadora Pão de Ontem - Tabela de preços
<br>1 - RS 0.18
<br>2 - RS 0.36
<br>...
<br>50 - RS 9.00

In [38]:
preco = float(input("Insira o preço do pão: "))
for num in range(1, 51):
    total = num * preco
    print("%s - R$ %.2f" % (num, total))

Insira o preço do pão: 0.20
1 - R$ 0.20
2 - R$ 0.40
3 - R$ 0.60
4 - R$ 0.80
5 - R$ 1.00
6 - R$ 1.20
7 - R$ 1.40
8 - R$ 1.60
9 - R$ 1.80
10 - R$ 2.00
11 - R$ 2.20
12 - R$ 2.40
13 - R$ 2.60
14 - R$ 2.80
15 - R$ 3.00
16 - R$ 3.20
17 - R$ 3.40
18 - R$ 3.60
19 - R$ 3.80
20 - R$ 4.00
21 - R$ 4.20
22 - R$ 4.40
23 - R$ 4.60
24 - R$ 4.80
25 - R$ 5.00
26 - R$ 5.20
27 - R$ 5.40
28 - R$ 5.60
29 - R$ 5.80
30 - R$ 6.00
31 - R$ 6.20
32 - R$ 6.40
33 - R$ 6.60
34 - R$ 6.80
35 - R$ 7.00
36 - R$ 7.20
37 - R$ 7.40
38 - R$ 7.60
39 - R$ 7.80
40 - R$ 8.00
41 - R$ 8.20
42 - R$ 8.40
43 - R$ 8.60
44 - R$ 8.80
45 - R$ 9.00
46 - R$ 9.20
47 - R$ 9.40
48 - R$ 9.60
49 - R$ 9.80
50 - R$ 10.00


<b>22. O Sr. Manoel Joaquim expandiu seus negócios para além dos negócios de 1,99 e
agora possui uma loja de conveniências. Faça um programa que implemente uma
caixa registradora rudimentar. O programa deverá receber um número
desconhecido de valores referentes aos preços das mercadorias. Um valor zero deve
ser informado pelo operador para indicar o final da compra. O programa deve então
mostrar o total da compra e perguntar o valor em dinheiro que o cliente forneceu,
para então calcular e mostrar o valor do troco. Após esta operação, o programa
deverá voltar ao ponto inicial, para registrar a próxima compra. A saída deve ser
conforme o exemplo abaixo:</b><br>
<br>Lojas Tabajara
<br>Produto 1: RS 2.20
<br>Produto 2: RS 5.80
<br>Produto 3: RS 0
<br>Total: RS 9.00
<br>Dinheiro: RS 20.00
<br>Troco: RS 11.00
<br>...

In [39]:
total = 0
preco = -1
i = 0
while preco != 0:
    preco = float(input("Insira o preço do item (Digite '0' para finalizar a compra): "))
    total += preco
    i += 1
    if preco == 0:
        print(f"Total: R$ {total}")
        pago = float(input("Insira o valor a ser pago: "))
        print("Troco: R$ ", pago - total)
    else:
        print(f"Produto {i}: R$ {preco}")

Insira o preço do item (Digite '0' para finalizar a compra): 2
Produto 1: R$ 2.0
Insira o preço do item (Digite '0' para finalizar a compra): 1.5
Produto 2: R$ 1.5
Insira o preço do item (Digite '0' para finalizar a compra): 0
Total: R$ 3.5
Insira o valor a ser pago: 4
Troco: R$  0.5


<b>23. Foi feita uma estatística em cinco cidades brasileiras para coletar dados sobre
acidentes de trânsito. Foram obtidos os seguintes dados:</b><br>
<br>- Código da cidade;
<br>- Número de veículos de passeio (em 1999);
<br>- Número de acidentes de trânsito com vítimas (em 1999). Deseja-se saber:
<br>- Qual o maior e menor índice de acidentes de transito e a que cidade pertence;
<br>- Qual a média de veículos nas cinco cidades juntas;
<br>- Qual a média de acidentes de trânsito nas cidades com menos de 2.000 veículos de
passeio.

In [2]:
soma_veiculos = 0
soma_acidentes = 0
soma_acidentes_menos_dois_mil = 0
total_cidades_menos_dois_mil = 0

for i in range(0, 5):
    codigo = int(input('\nDigite o código da cidade: '))
    veiculos = int(input('Digite o número de veículos de passeio: '))
    acidentes = int(input('Digite a quantidade de acidentes : '))
    indice_acidentes = acidentes / float(veiculos)
    soma_veiculos += veiculos
    if ('mais_acidentes' not in vars()) or (indice_acidentes > mais_acidentes):
        mais_acidentes = indice_acidentes
        codigo_mais_acidentes = codigo
    if ('menos_acidentes' not in vars()) or (indice_acidentes < menos_acidentes):
        menos_acidentes = indice_acidentes
        codigo_menos_acidentes = codigo

    if (veiculos < 2000):
        soma_acidentes_menos_dois_mil += acidentes
        total_cidades_menos_dois_mil += 1

print ('\nA cidade com maior índice de acidentes é %i com %.2f' %(codigo_mais_acidentes, mais_acidentes))
print ('A cidade com menor índice de acidentes é %i com %.2f' %(codigo_menos_acidentes, menos_acidentes))
print ('A média de veículos nas cidades é %.2f' % (soma_veiculos / 5.0))
print ('A média de acidentes de trânsito nas cidades com menos de dois mil veículos é %.2f' %(soma_acidentes_menos_dois_mil / float(total_cidades_menos_dois_mil)))


Digite o código da cidade: 125
Digite o número de veículos de passeio: 466
Digite a quantidade de acidentes : 478

Digite o código da cidade: 413
Digite o número de veículos de passeio: 36544
Digite a quantidade de acidentes : 8465

Digite o código da cidade: 6631
Digite o número de veículos de passeio: 35
Digite a quantidade de acidentes : 3641

Digite o código da cidade: 2145
Digite o número de veículos de passeio: 63
Digite a quantidade de acidentes : 215

Digite o código da cidade: 45
Digite o número de veículos de passeio: 21
Digite a quantidade de acidentes : 1

A cidade com maior índice de acidentes é 6631 com 104.03
A cidade com menor índice de acidentes é 45 com 0.05
A média de veículos nas cidades é 7425.80
A média de acidentes de trânsito nas cidades com menos de dois mil veículos é 1083.75


<b>24. Faça um programa que receba o valor de uma dívida e mostre uma tabela com
os seguintes dados: valor da dívida, valor dos juros, quantidade de parcelas e valor
da parcela.</b><br>
Os juros e a quantidade de parcelas seguem a tabela abaixo:<br>
<br>Quantidade de Parcelas % de Juros sobre o valor inicial da dívida
<br>1 0
<br>3 10
<br>6 15
<br>9 20
<br>12 25<br>
<br>Exemplo de saída do programa:<br>
<br>Valor da Dívida Valor dos Juros Quantidade de Parcelas Valor da Parcela
<br>RS 1.000,00 0 1 RS 1.000,00
<br>RS 1.100,00 100 3 RS 366,00
<br>RS 1.150,00 150 6 RS 191,67

In [3]:
dividaInicial = float(input("Insira o valor da dívida: "))
print("Dívida \t Valor Juros \t Qtde Parcelas \t Valor da Parcela")
for i in range(0, 5):
    if i == 0:
        juros = 0
    elif i == 1:
        juros = 10
    else:
        juros += 5
    valorJuros = dividaInicial * (juros / 100)
    divida = dividaInicial + valorJuros
    if i == 0:
        qtdeParc = 1
    else:
        qtdeParc = i * 3
    if qtdeParc != 0:
        valorParcela = divida / qtdeParc
    else:
        valorParcela = divida
    print(f"R$ {divida} \t {valorJuros} \t {qtdeParc} \t R$ %.2f"%(valorParcela))

Insira o valor da dívida: 300
Dívida 	 Valor Juros 	 Qtde Parcelas 	 Valor da Parcela
R$ 300.0 	 0.0 	 1 	 R$ 300.00
R$ 330.0 	 30.0 	 3 	 R$ 110.00
R$ 345.0 	 45.0 	 6 	 R$ 57.50
R$ 360.0 	 60.0 	 9 	 R$ 40.00
R$ 375.0 	 75.0 	 12 	 R$ 31.25


<b>25. Faça um programa que mostre os n termos da Série a seguir:</b><br>
<br> S = 1/1 + 2/3 + 3/5 + 4/7 + 5/9 + ... + n/m.<br>
Imprima no final a soma da série.

In [4]:
termo = int(input('Insira a quantidade de termos desejada: '))

somaStr = "S = "
s = 0
m = 1
for n in range(1, termo + 1):
    s += n / m
    somaStr += str(n) + "/" + str(m)
    if n != termo:
        somaStr += " + "
    m += 2

print(somaStr)
print('Soma da série: %s' % s)

Insira a quantidade de termos desejada: 30
S = 1/1 + 2/3 + 3/5 + 4/7 + 5/9 + 6/11 + 7/13 + 8/15 + 9/17 + 10/19 + 11/21 + 12/23 + 13/25 + 14/27 + 15/29 + 16/31 + 17/33 + 18/35 + 19/37 + 20/39 + 21/41 + 22/43 + 23/45 + 24/47 + 25/49 + 26/51 + 27/53 + 28/55 + 29/57 + 30/59
Soma da série: 16.34118842374577
